crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -166,29 +186,38 @@ impl Object {
             .extend()
     }
 
-    /// Objects can either be immutable, shared, owned by an address,
-    /// or are child objects (part of a dynamic field)
-    async fn kind(&self) -> Option<ObjectKind> {
+    /// The owner type of this Object.
+    /// Immutable and Shared Objects do not have owners.
+    async fn owner(&self, ctx: &Context<'_>) -> Option<ObjectOwnerNew> {
         use NativeOwner as O;
-        use ObjectKind as K;
-        Some(match self.native.owner {
-            O::AddressOwner(_) => K::Owned,
-            O::ObjectOwner(_) => K::Child,
-            O::Shared { .. } => K::Shared,
-            O::Immutable => K::Immutable,
-        })
-    }
 
-    /// The Address or Object that owns this Object.  Immutable and Shared Objects do not have
-    /// owners.
-    async fn owner(&self) -> Option<Owner> {
-        use NativeOwner as O;
-        let (O::AddressOwner(address) | O::ObjectOwner(address)) = self.native.owner else {
-            return None;
-        };
+        if self.native.owner.is_child_object() {
+            let parent = match self.native.owner.get_owner_address().ok() {
+                Some(addr) => ctx
+                    .data_unchecked::<PgManager>()
+                    .fetch_obj(addr.into(), None)
+                    .await
+                    .ok()
+                    .flatten(),
+                None => None,
+            };
+            return Some(ObjectOwnerNew::Parent(Parent { parent }));
+        }
 
-        let address = SuiAddress::from(address);
-        Some(Owner { address })
+        match self.native.owner {
+            O::AddressOwner(address) | O::ObjectOwner(address) => {
```

Comment:
fyi self.native.owner matches O::ObjectOwner == self.native.owner.is_child_object(), you've already done the check for above. Might be better to roll everything into the same match?

Reply:
Yes please!

Reply:
This syntax really confuses me...